### DISCUSSION TITLE
How to "unwrap" each element in a readonly tuple?

### DISCUSSION THREAD

**Discussion by samsze0** [type-challenges/type-challenges#29070] [Q&A]
```typescript
type ExtractType<T extends StoreApi<any>> = T extends StoreApi<infer U>
  ? U
  : never;

type UnwrapStoreApi<T extends StoreApi<any>[]> = T extends [
  infer Head,
  ...infer Tail
]
  ? readonly [ExtractType<Head>, ...UnwrapStoreApi<Tail>]
  : readonly [];
```

Problem: I want to write a type function that extract the inner type of each element in a tuple. The error that I am encountering: the result of `UnwrapStoreApi` always equals to `readonly []`

**Comment by jazelly** [comment]
Can I see your `StoreApi` type? If it is readonly, your UnwrapStoreApi will be evaluated to `readonly []`, because tuple extends readonly tuple, but not vice versa.
